### DISCUSSION TITLE
Strange tag list results.

### DISCUSSION THREAD

**Discussion by rkrupnek** [ottowayi/pycomm3#266] [Q&A]
First off, thanks for this great library!

I'm working through the docs, and trying the Tag List example, I get some strange results.  I don't know if you have done any testing with Python 3.11 yet, but I tested this with 3.11.1 and 3.11.2 and got the same weirdness.  So, I'm not sure if it's due to something in the pycomm3 library or something that got changed in Py3.11.

Nonetheless, here is what I tried:

```
def tag_list_equal():
    with LogixDriver(PLC_IP) as plc1:
        tag_list = plc1.get_tag_list()
        if {tag['tag_name']: tag for tag in tag_list} == plc1.tags:
            print('plc1 get_tag_list() result is the same as plc1.tags')

    with LogixDriver(PLC_IP, init_tags=False) as plc2:
        plc2.get_tag_list()

    if plc1.tags == plc2.tags:
        print('Calling get_tag_list() does the same thing.')
    else:
        print('Calling get_tag_list() does NOT do the same thing.')
        diff = { k : plc2.tags[k] for k in set(plc2.tags) - set(plc1.tags) }
        if not diff:
            print('But really they ARE the same thing.')
```

And here was my result...
```
plc1 get_tag_list() result is the same as plc1.tags
Calling get_tag_list() does NOT do the same thing.
But really they ARE the same thing.
```

Am I missing something?

**Comment by rkrupnek** [comment]
FWIW, I just tried in a 3.10.10  venv as well as a 3.7.7 venv, and it seems to do the same thing.

**Reply by ottowayi** [reply]
I'm betting that's because there's some objects (probably the data types, but I haven't looked at the code) that don't support the dunder-equal methods so it falls back to comparing the `id(obj)` which will be false b/c they're different objects.  Can you try comparing the `tags_json` property instead of `tags`?  That one strips out some the non-json-serializable fields and should be equal.

**Comment by rkrupnek** [comment]
I was kind-of wondering if that wasn't the case.  It threw me off when I got different results than the examples showed.  But using the tags_json property does compare equal.  I'm just getting started with the library, so thanks for the quick response!
